src/test/java/com/google/sps/servlets/TestCredVerifierServlet.java

```diff
@@ -0,0 +1,77 @@
+package com.google.sps.servlets;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.BearerToken;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.Rule;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class TestCredVerifierServlet {
+  Credential mockCredential;
+  StringWriter stringWriter;
+  PrintWriter writer;
+  @Mock
+  AuthorizationCodeFlow mockFlow;
+  @Mock
+  UserService mockUserService;
+  @Mock
+  HttpServletRequest mockRequest;
+  @Mock
+  HttpServletResponse mockResponse;
+
+  @Before
+  public void setUp() throws IOException {
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    mockCredential = new Credential(BearerToken.authorizationHeaderAccessMethod());
+    when(mockUserService.getCurrentUser()).thenReturn(new User("email", "authDomain", "1234"));
+    when(mockFlow.loadCredential("1234")).thenReturn(mockCredential);
```

Comment:
I think this is essential to understand the tests. Maybe just create the user in the setup and then move this to your tests (I know it is duplicated code, but for tests it is more important that you can understand them by reading each single test method than writing compact code).